okhttp-testing-support/src/main/java/okhttp3/testing/PlatformRule.kt

```diff
@@ -50,12 +86,54 @@ open class PlatformRule @JvmOverloads constructor(
     }
   }
 
-  override fun after() {
+  fun resetPlatform() {
     if (platform != null) {
       Platform.resetForTests()
     }
   }
 
+  @JvmOverloads
+  fun expectFailure(
```

Comment:
This feels like too much framework just like JUnits bad expected exceptions feature. 

Just use try/catch? Then you can isolate the expectation properly, and assert the state of things after the exception if you need to.

Reply:
Truth be told it’s less about how it’s failing. More about understanding what is failing in all the weird combinations. So I think I’ll drop the how aspect and just focus on when. So we detect that a conscrypt release or JDK 11.0.4 fixed it.

Reply:
But I also want to keep the test focused on the correct case. So ideally without an embedded try catch check env. And a time-out is thrown externally so doesn’t work for that.

Reply:
Okay.